vars/neoDeploy.groovy

```diff
@@ -216,11 +231,80 @@ void call(parameters = [:]) {
                 lock("$STEP_NAME:${neoCommandHelper.resourceLock()}") {
                     deploy(script, configuration, neoCommandHelper, configuration.dockerImage, deployMode)
                 }
+                if(configuration.neo.invalidateCache == true) {
+                    inValidateCache(configuration)
+                }
             }
         }
     }
 }
 
+private inValidateCache(configuration){
+    def account = configuration.neo.account
+    def host = configuration.neo.host
+
+    withCredentials([usernamePassword(
+        credentialsId: configuration.neo.oauthCredentialId,
+        passwordVariable: 'OAUTH_NEO_CLIENT_SECRET',
+        usernameVariable: 'OAUTH_NEO_CLIENT_ID')]) {
+        def bearerTokenResponse = sh(
+            script: """#!/bin/bash
+                 curl -X POST -u "${OAUTH_NEO_CLIENT_ID}:${OAUTH_NEO_CLIENT_SECRET}" \
+                 --fail \
+           \"https://oauthasservices-${account}.${host}/oauth2/api/v1/token?grant_type=client_credentials&scope=write,read\"
+                """,
+                returnStdout: true
+            )
+        def fetchBearerTokenJson = readJSON  text: bearerTokenResponse
+
+        def bearerToken = fetchBearerTokenJson.access_token
+
+        echo "Retrieved bearer token."
+
+        def xcsrfTokenResponse = sh(
+            script: """#!/bin/bash
+                              curl -i -L \
+                              -c 'cookies.jar' \
+                              -H 'X-CSRF-Token: Fetch' \
+                              -H "Authorization: Bearer ${bearerToken}" \
+                              --fail \
+                           \"https://sandboxportal-${account}.${host}/fiori/api/v1/csrf\"
+                    """, returnStdout: true)
+
+        def xcsrfTokenHeaderMatcher=xcsrfTokenResponse =~ /(?m)^X-CSRF-Token: ([0-9A-Z]*)$/
+        def xcsrfToken = xcsrfTokenHeaderMatcher[0][1]
+
+        def siteId = configuration.neo.siteId ? "${configuration.neo.siteId}" : ""
+        if(siteId ==""){
+            echo "Using the default value of siteId from Portal service and invalidating the cache."
+        }
+        else{
+                echo "Invalidating the cache for site Id : ${siteId}."
+        }
+        def statusCode = sh(
+            script: """#!/bin/bash
+                      curl -X POST -L \
+                      -b 'cookies.jar'  \
+                      -H "X-CSRF-Token: ${xcsrfToken}" \
+                      -H "Authorization: Bearer ${bearerToken}" \
+                      -d "{\"siteId\":${siteId}}" \
+                      -so /dev/null \
+                      -w '%{response_code}' \
+                      --fail \
+                      \"https://sandboxportal-${account}.${host}/fiori/v1/operations/invalidateCache\"
+                  """,
+            returnStdout: true).trim()
+
+        if (siteId=="" && statusCode == "500") {
+            error "Invalidating the cache failed with status code: ${statusCode}. " +
```

Comment:
I don't think we will reach this line of code in case of a 500 error, as the step will fail in the sh curl call. Unless it is prevented by the `-so /dev/null`. In that case the other error codes need to be handled as well and through an error. We don't want it to say "Successfully invalidated the cache" after a 401 error, or whatever.